flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -791,6 +796,52 @@ public TableResult executePlan(CompiledPlan plan) {
         return executeInternal(transformations, sinkIdentifierNames);
     }
 
+    private CompiledPlan compilePlanAndWrite(
+            String filePath, boolean ifNotExists, Operation operation) {
+        File file = Paths.get(filePath).toFile();
+        if (file.exists()) {
+            if (ifNotExists) {
+                try {
+                    return loadPlan(PlanReference.fromFile(filePath));
+                } catch (IOException e) {
+                    throw new TableException("Cannot load the plan file '" + filePath + "'", e);
```

Comment:
TBH SQL should be a thin wrapper around Table API. We can also use `TableException` for `loadPlan` and make the nice `Cannot load plan file...` for all users consistently. Checked exceptions are annoying anyway and it should be obvious that a lot of stuff can go wrong (not only IO with files but also catalog etc.).